pinot-core/src/main/java/org/apache/pinot/core/transport/server/routing/stats/ServerRoutingStatsManager.java

```diff
@@ -171,6 +183,7 @@ public void recordStatsUponResponseArrival(long requestId, String serverInstance
 
     _executorService.execute(() -> {
       try {
+        alertIfQueueSizeIsAboveWarnThreshold();
```

Comment:
Might not be necessary to collect stats at both points -> query submission and query response.  Only submission should be sufficient.

Reply:
addressed. Only emit metrics after submission.